content/open-source/wolfi/building-a-wolfi-package.md

```diff
@@ -0,0 +1,437 @@
+- Other
+  - `strip`
+  - `patch`
+
+Each pipeline can have one or more parameters that should be provided as keypairs in a `with` entry. For example, this is how a download-and-check step looks like in the melange YAML, using the built-in pipeline `fetch`:
+
+```yaml
+  - uses: fetch
+    with:
+      uri: https://libzip.org/download/libzip-${{package.version}}.tar.gz
+      expected-sha256: 52a60b46182587e083b71e2b82fcaaba64dd5eb01c5b1f1bc71069a3858e40fe
+```
+
+Naturally, you can also run raw bash commands in your pipeline. The following example shows the build of [Composer](https://getcomposer.org/), the PHP package manager, which is not compiled as a typical package:
+
+```yaml
+  - name: Install Composer
+    runs: |
+      EXEC_DIR="${{targets.destdir}}/usr/bin"
+      mkdir -p "${EXEC_DIR}"
+      mv composer.phar "${EXEC_DIR}/composer"
+      chmod +x "${EXEC_DIR}/composer"
+```
+
+As indicated, a pipeline step will have either a `uses` or a `run` directive. You can have as many steps as you need, and you can use special variable substitutions inside steps:
+
+- `${{package.name}}`: Package name.
+- `${{package.version}}`: Package version.
+- `${{package.epoch}}`: Package epoch.
+- `${{targets.destdir}}`: Directory where final package files will be stored. Everything that lives here will be packed into your final apk.
+- `${{targets.subpkgdir}}`: Directory where final subpackage files will be stored. Works the same way as `targets.destdir`, but for subpackages.
+
+You can find more details about available pipelines in the [melange pipelines documentation](https://edu.chainguard.dev/open-source/melange/melange-pipelines).
+
+### The `subpackages` Section
+As mentioned above, a package may extract parts of its contents into subpackages, in order to make for a slimmer final apk. Many packages have resources that are not required at execution time: development headers, man pages, shared libraries that are optional… you name it. This part is really important in Wolfi, because we want packages to be minimal. The `subpackages` section of the melange YAML file looks a lot like the pipeline section, and it essentially works the same way. You'll just have to make sure you place any subpackage files in the `targets.subpkgdir` location.
+
+The `split` built-in pipelines were created to facilitate the creation of subpackages. They implement code to remove development headers (`split/dev`), man pages (`split/manpages`), among other resources that aren't typically required at runtime. You can experiment with those, just be aware that they use standard path locations and some compiled packages may use different paths for certain resources.
+
+For example, this is how a step in the subpackages section would be written, using the `split/dev` built-in pipeline to generate the `php-dev` subpackage:
+
+```yaml
+  - name: php-dev
+    description: PHP 8.2 development headers
+    pipeline:
+      - uses: split/dev
+```
+
+#### Looping with Ranges
```

Comment:
```suggestion
**Looping with Ranges**
```

I'm really not sure this is a better solution but I don't think we've properly configured h4's. On the preview at least, this h4 looks bigger than the h3s